### DISCUSSION TITLE
How to sort by a nested property?

### DISCUSSION THREAD

**Discussion by MilesAdamson** [isar/isar#1215] [Help]
Goal: Sort a Job by activity id, which lives in a big nested thing called `ticketUnitNumber`,

Creating a `SortProperty` with `ticketUnitNumber.activityId` and `activityId` doesn't work. Not seeing a way in the docs to do this

![image](https://user-images.githubusercontent.com/29047682/234400594-09935379-146e-428d-9b03-9391e1eb6437.png)

**Comment by simc** [comment]
That's not possible unfortunately. I would create a getter in the root object and expose the value to isar. Then you can use it for sorting.

**Reply by MilesAdamson** [reply]
Ok so if `Job` just has:

`int get activityId => ticketUnitNumber.activityId`

I should be able to sort by it? Can that be indexed still?

**Reply by simc** [reply]
Exactly. You can sort and index by that getter
